### DISCUSSION TITLE
SDL 2.26.3 LINK : fatal error LNK1561

### DISCUSSION THREAD

**Discussion by PeakRacing** [xmake-io/xmake#3451] [Q&A] [answered]
您好，sdl包使用2.26.3版本编译会报LINK : fatal error LNK1561: 必须定义入口点
![image](https://user-images.githubusercontent.com/24759100/222665373-b9f7c5f1-0d48-441c-b085-873b0f034e85.png)
请问需要改哪里呢？

**Comment by waruqi** [comment, marked as answer]
main 如果在 static 里面，得额外自己设置 /subsystem:console 的 ldflags

**Reply by PeakRacing** [reply]
main 并不在 static 里，2.26.3之前的版本编译无问题，只有2.26.3报这个

**Reply by PeakRacing** [reply]
> main 如果在 static 里面，得额外自己设置 /subsystem:console 的 ldflags

msvc llvm都试过，一样的结果

**Reply by waruqi** [reply]
没 link 上 sdlmain ，你调下包 https://github.com/xmake-io/xmake-repo/blob/master/packages/l/libsdl/xmake.lua

**Reply by PeakRacing** [reply]
> 没 link 上 sdlmain ，你调下包 https://github.com/xmake-io/xmake-repo/blob/master/packages/l/libsdl/xmake.lua

找到原因了，用包 https://github.com/xmake-io/xmake-repo/blob/master/packages/l/libsdl/xmake.lua 编译正常，但是github有的人访问困难，我采用了 set_base方式继承修改

package("nes_libsdl")
    set_base("libsdl")
    set_urls("https://www.libsdl.org/release/SDL2-2.26.3.zip", { alias = "official" })
    add_versions("official:2.26.3", "3e46df1eb1be30448cf7c7f3fc0991980f0ef867c2412ab7bc925b631e5dc09c")
package_end()
add_requires("nes_libsdl 2.26.3", {system = false, alias = "libsdl"})
add_packages("libsdl")

此方式就会报错，大佬看下哪里使用错误

**Reply by PeakRacing** [reply]
> 没 link 上 sdlmain ，你调下包 https://github.com/xmake-io/xmake-repo/blob/master/packages/l/libsdl/xmake.lua

改了改还是不行，测试使用set_base方式还是会失败

**Reply by waruqi** [reply]
再试下，应该可以了，之前的包也要重装下

```
xmake update -s dev
xrepo remove --all nes_libsdl
xmake f -c
xmake
```

**Reply by PeakRacing** [reply]
> 再试下，应该可以了，之前的包也要重装下
> 
> ```
> xmake update -s dev
> xrepo remove --all nes_libsdl
> xmake f -c
> xmake
> ```

测试可以了，感谢大佬
